plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergModule.java

```diff
@@ -96,4 +100,12 @@ public void configure(Binder binder)
         tableProcedures.addBinding().toProvider(ExpireSnapshotsTableProcedure.class).in(Scopes.SINGLETON);
         tableProcedures.addBinding().toProvider(RemoveOrphanFilesTableProcedure.class).in(Scopes.SINGLETON);
     }
+
+    @Singleton
+    @Provides
+    @ForMetadataFetching
+    public ForkJoinPool createIcebergMetadataFetchingForkJoinPool(IcebergConfig icebergConfig)
+    {
+        return new ForkJoinPool(icebergConfig.getMaxConcurrentMetadataLoaders());
```

Comment:
We could consider providing a configuration that would restore previous behavior of "one thread per query".
i think we may already have cases like that in Glue metastore, we can inspect try to follow existing pattern (if found)